docs/compatibility.html

```diff
@@ -0,0 +1,206 @@
+    <p>Note: Can’t upgrade server from static voter to dynamic voter, see <a href="https://issues.apache.org/jira/browse/KAFKA-16538">KAFKA-16538</a>.</p>
+
+    <h4>Client Compatibility</h4>
+    <table>
+        <tr>
+            <th>Kafka Version</th>
+            <th>Module</th>
+            <th>Compatibility with Kafka 4.0</th>
+            <th>Key Differences/Limitations</th>
+        </tr>
+        <tr>
+            <td rowspan="3">0.x</td>
+            <td>Client</td>
+            <td>❌ Not Compatible</td>
+            <td>Pre-0.10.x protocols are fully removed in Kafka 4.0 (<a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-896">KIP-896</a>).</td>
+        </tr>
+        <tr>
+            <td>Streams</td>
+            <td>❌ Not Available</td>
+            <td>Kafka Streams introduced in Kafka 0.10.x.</td>
+        </tr>
+        <tr>
+            <td>Connect</td>
+            <td>⚠️ Limited Compatibility</td>
+            <td>Connect introduced in Kafka 0.9.x but lacks support for modern features like IncrementalAlterConfigs and dynamic configs.</td>
+        </tr>
+        <tr>
+            <td rowspan="3">1.x</td>
+            <td>Client</td>
+            <td>⚠️ Limited Compatibility</td>
+            <td>Supports brokers >= 0.10.x. Does not support newer features like idempotent producer or transactional API.</td>
+        </tr>
+        <tr>
+            <td>Streams</td>
+            <td>⚠️ Limited Compatibility</td>
+            <td>Basic Streams features available. Exactly-once semantics unavailable (requires Kafka 2.4.x or later).</td>
+        </tr>
+        <tr>
+            <td>Connect</td>
+            <td>⚠️ Limited Compatibility</td>
+            <td>IncrementalAlterConfigs not supported before Kafka 2.3.x.</td>
+        </tr>
+        <tr>
+            <td rowspan="3">2.x</td>
+            <td>Client</td>
+            <td>⚠️ Partially Compatible</td>
+            <td>Compatible with brokers >= 0.10.x but lacks optimizations for Kafka 4.0. Pre-0.10.x protocols removed in 4.0.</td>
+        </tr>
+        <tr>
+            <td>Streams</td>
+            <td>⚠️ Limited Compatibility</td>
+            <td>Exactly-once semantics unavailable (requires Kafka 2.4.x or later). Pre-2.4.x brokers lack key Streams features.</td>
+        </tr>
+        <tr>
+            <td>Connect</td>
+            <td>⚠️ Limited Compatibility</td>
+            <td>MirrorMaker 1 supported in 2.x but removed in 4.0. IncrementalAlterConfigs unsupported before Kafka 2.3.x.</td>
+        </tr>
+        <tr>
+            <td rowspan="3">3.x</td>
+            <td>Client</td>
+            <td>✅ Fully Compatible</td>
+            <td>Fully compatible with Kafka 4.0 brokers. Lacks access to optimizations introduced in Kafka 4.0.</td>
+        </tr>
+        <tr>
+            <td>Streams</td>
+            <td>✅ Fully Compatible</td>
+            <td>Supports exactly-once semantics (introduced in Kafka 2.4.x). Topic IDs introduced in Kafka 3.0.</td>
```

Comment:
EOS was added in KS 0.11.0. -- We did add EOSv2 in KS 2.6 (which required 2.5.x broker), but given that we talk about KS 3.x against 4.0 brokers it should not matter.

Why would we care about Topic IDs?